iframe/priv/static/iframe/v3.html

```diff
@@ -80,6 +81,11 @@
           handleSync(callback) {
             state.syncHandler = callback;
           },
+
+          requestSecret(currentSecret, callback) {
```

Comment:
I would just keep the name `currentSecret` here, as we are passing the currently selected secret. Since we are not passing the options and we don’t have a default, it seems a bit confusing to me. WDYT?

Reply:
It seems valid to do `selectSecret((secretLabel) => { ... })`. In such case we just don't prefill anything and we get the user selects/creates. I'm not saying it's necessarily more useful, but `currentSecret` doesn't seem required, so it doesn't hurt to have it as an option. Does that make sense?

Reply:
Hey! I had to change a bit here. Now we're handling unavailable secrets in the modal, which makes `currentSecret` required. But we can also receive more information optionally. @josevalim has some ideas around this!

Reply:
When the secret field is empty we would pass `currentSecret = ""`, since there isn't really a current secret at that point. I don't what's wrong with `selectSecret((secretLabel) => { ... })`, the user would either pick one of the existing secrets or create a new one with whichever any name they want.

The secret name would only be required (moreover fixed) when we show the error "we didn't find this paticular secret, please create or allow", no?

Reply:
That works too and I’ve made the change, however, I keep my opinion that it should be explicitly required since the experience we are designing depends on that value to be consistent.

Reply:
I agree we should force the current secret to be given, even if empty string, because it is very important for the flow of the modal.

Reply:
> The secret name would only be required (moreover fixed) when we show the error "we didn't find this paticular secret, please create or allow", no?

We need the current secret for:

1. If it exists, highlight it is currently the one selected
2. If it does not exist, prefill the "Name" field with the secret value

Forgetting to pass it will lead to a broken experience IMO.

Reply:
We can always pass an empty string, so if you both feel strongly about keeping it as argument, it's fine for me. I would rename it to `preselectName` or similar.